### PR TITLE
Use the accessor names as written in Objective-C.

### PR DESCRIPTION
**Pull request by varungandhi-apple**
Not 100% sure if my approach taken here is correct. It is unclear to me what the semantics of `getParsedAccessor` are supposed to be in the presence of user-written accessor names in Objective-C (as opposed to those written in Swift). Should they be passed through? Or not passed through because they are considered implicit? I've assumed that the existing behavior of `getParsedAccessor` is correct, so we use `getAccessor` instead.

Fixes rdar://problem/55519276.

Related: commit 0c5d52d8609aaa3016621887d03571c7582984a5 (PR #26461) where `getParsedAccessor` was introduced.

### CODE REVIEW

**Comment by varungandhi-apple** [comment]
@swift-ci please test

File: test/PrintAsObjC/getter_setter.swift [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+import Foundation
+
+// rdar://problem/55519276: Make sure we're using the right selector names as
+// written in Objective-C.
+public final class Saiyan: NSObject, PowerProtocol {
+  public var number: Int64
+  // CHECK: @property (nonatomic, getter=getPower, setter=setPower:) int64_t number;
+  public override init() {
+    number = 9001
```
Comment (Azoy) [inline test/PrintAsObjC/getter_setter.swift]: (It’s over 9000!?!)

**Comment by varungandhi-apple** [comment]
@swift-ci please test

**Comment by rjmccall** [comment]
IIRC, Slava was aiming to eliminate uses of the raw `getAccessor` method.  I think `getParsedAccessor` is meant to be solely about what was written in source, so it's reasonable to say that we shouldn't be using it in this semantic method, but it's not clear to me that there's currently a better method, since we (apparently) can't assert as a precondition that this is an `@objc` property.  Slava would have thoughts.

**Comment by slavapestov** [comment]
I think the right fix is to not mark imported accessors that correspond to objc methods as implicit. Only bona fide synthesized accessors should be implicit (eg, those for stored properties of structs, or a modify accessor synthesized when a property witnesses a mutable protocol requirement, etc).

**Comment by varungandhi-apple** [comment]
@swift-ci please test Linux platform

**Comment by varungandhi-apple** [comment]
@swift-ci please test

**Comment by varungandhi-apple** [comment]
Note: I've added a new commit as I think @edymtt cherry-picked the first two commits elsewhere, so I didn't want to mess with history by squashing the third commit on top of the second one, just in case it causes issues.

**Comment by varungandhi-apple** [comment]
Note for anyone else reading: the imported accessor is explicit in the existing code. Which is correct behavior. Hence, I didn't make any changes there.

**Pull request merged**